lib/matplotlib/backends/backend_ps.py

```diff
@@ -954,8 +955,37 @@ def print_figure_impl(fh):
                         fh.write(_font_to_ps_type3(font_path, glyph_ids))
                     else:
                         try:
-                            convert_ttf_to_ps(os.fsencode(font_path),
-                                              fh, fonttype, glyph_ids)
+                            _log.debug(
+                                "SUBSET %s characters: %s", font_path,
+                                ''.join(chr(c) for c in chars)
+                            )
+                            fontdata = _backend_pdf_ps.get_glyphs_subset(
+                                font_path, "".join(chr(c) for c in chars)
+                            )
+                            _log.debug(
+                                "SUBSET %s %d -> %d", font_path,
+                                os.stat(font_path).st_size,
+                                fontdata.getbuffer().nbytes
+                            )
+
+                            # give ttconv a subsetted font
+                            # along with updated glyph_ids
+                            with tempfile.NamedTemporaryFile(
+                                suffix=".ttf"
+                            ) as tmp:
+                                font = FT2Font(fontdata)
+                                glyph_ids = [
+                                    font.get_char_index(c) for c in chars
+                                ]
+                                tmp.write(fontdata.getvalue())
```

Comment:
I think we should add a `tmp.flush()` to avoid a race condition between the content being written out and `convert_ttf_to_ps()` reading it back.

Reply:
I was under the impression that Python interpreter will only move to the next line once the `write` is complete?

Seems like `.flush()` 'clears out the internal buffer of the file', which is still in memory here, but once we've written it out on disk, could a race condition still exist?

(asking totally on the learning side here, I can add this in my next commit)

Reply:
Write operations on a file object in Python are usually [buffered](https://docs.python.org/3/library/functions.html#open) with a [default buffer size](https://docs.python.org/3/library/io.html#io.DEFAULT_BUFFER_SIZE) of 4KiB or 8KiB. Calling `flush()` ensures that another process can read back the content, [see](https://stackoverflow.com/questions/7127075/what-exactly-is-pythons-file-flush-doing). The OS might still buffer and delay the write operation to the physical disk for performance reasons. This is completely transparent to user programs. To ensure that the data is permanently written to disk and persisted in case of a system crash, you could call `fsync()` but this is not necessary in this case.

Reply:
Thank you so much for a detailed answer! (that SO link was an amazing and insightful read)

Reply:
The context manager calls `.close`, which flushes the buffer; there's no need for an explicit `.flush` here (at least in how the code ended up as merged.)